projects/igniteui-angular/src/lib/badge/badge.component.ts

```diff
@@ -173,46 +164,30 @@ export class IgxBadgeComponent {
         return this.type + ' badge type without value';
     }
 
-    /**
-     * Method which makes the name of the class more descriptive.
-     * This helps the styling of the badges.
-     * @hidden
-     * @internal
-     */
-    @HostBinding('class')
-    get setClasses() {
-        let classes = {};
-
-        switch (IgxBadgeType[this.type.toUpperCase()]) {
-            case IgxBadgeType.INFO:
-                classes = {
-                    [`${this.cssClass}__circle--info`]: true
-                };
-                break;
-            case IgxBadgeType.SUCCESS:
-                classes = {
-                    [`${this.cssClass}__circle--success`]: true
-                };
-                break;
-            case IgxBadgeType.WARNING:
-                classes = {
-                    [`${this.cssClass}__circle--warning`]: true
-                };
-                break;
-            case IgxBadgeType.ERROR:
-                classes = {
-                    [`${this.cssClass}__circle--error`]: true
-                };
-                break;
-            default:
-                classes = {
-                    [`${this.cssClass}__circle--default`]: true
-                };
-        }
+    @HostBinding('class.igx-badge--default')
```

Comment:
No need for the `--default` modifier. Also, that doesn't need to be a getter as it never changes. It could simply be a string like I mentioned in my previous review. A user calling badge.defaultCSS will get `true`, which is not helpful. Take a look at how this is done in other components.